DataFormats/SiStripCluster/interface/SiStripApproximateClusterCollection.h

```diff
@@ -43,9 +43,10 @@ class SiStripApproximateClusterCollection {
     DetSet(SiStripApproximateClusterCollection const* coll, unsigned int detIndex)
         : coll_(coll),
           detIndex_(detIndex),
-          clusBegin_(coll_->beginIndices_[detIndex]),
-          clusEnd_(detIndex == coll_->beginIndices_.size() - 1 ? coll->clusters_.size()
-                                                               : coll_->beginIndices_[detIndex + 1]) {}
+          clusBegin_(coll_->beginIndices_.empty() ? 0 : coll_->beginIndices_[detIndex]),
+          clusEnd_(coll_->beginIndices_.empty()                  ? 0
```

Comment:
Using `coll->clusters_.size()` as the special index instead of `0` would give behavior that is closer to an empty `std::vector` (i.e. de-referencing the `DetSet::begin()` iterator is truly undefined behavior)

Reply:
I'm not sure I understand this comment; if `beginIndices_` is empty, then `clusters_.size()` is 0, so what's the difference?

Reply:
Ah right, the case here is that the whole collection is empty? How does one get a valid `DetSet` proxy object then? Perhaps it would be better to change `SiStripApproximateClusterCollection::begin()` to return `end()` when the collection is empty?

Reply:
That is indeed simpler and better. Updated the branch accordingly.